navigation/navigation-common/src/main/java/androidx/navigation/NavType.java

```diff
@@ -317,8 +317,7 @@ public Integer get(@NonNull Bundle bundle, @NonNull String key) {
         @NonNull
         @Override
         public Integer parseValue(@NonNull String value) {
-            throw new UnsupportedOperationException(
-                    "References don't support parsing string values.");
+            return Integer.parseInt(value);
```

Comment:
Let's match the implementation of `IntType` and ensure we can also handle hex encoded values as well. Write a test for this case as well

Reply:
done in 4d26376